src/Activities/Internal/EventSource/ActivityEventSender.cs

```diff
@@ -6,25 +6,28 @@
 using System.Diagnostics;
 using System.Globalization;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Microsoft.Omex.Extensions.Abstractions;
 using Microsoft.Omex.Extensions.Abstractions.Activities;
 using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
 using Microsoft.Omex.Extensions.Abstractions.ExecutionContext;
+using Microsoft.Omex.Extensions.Abstractions.Option;
 
 namespace Microsoft.Omex.Extensions.Activities
 {
 	internal sealed class ActivityEventSender : IActivitiesEventSender
 	{
-		public ActivityEventSender(ActivityEventSource eventSource, IExecutionContext executionContext, ILogger<ActivityEventSender> logger)
+		public ActivityEventSender(ActivityEventSource eventSource, IExecutionContext executionContext, ILogger<ActivityEventSender> logger, IOptions<MonitoringOption> options)
 		{
 			m_eventSource = eventSource;
 			m_serviceName = executionContext.ServiceName;
 			m_logger = logger;
+			m_options = options;
 		}
 
 		public void SendActivityMetric(Activity activity)
 		{
-			if (!m_eventSource.IsEnabled())
+			if (m_options.Value.DisableActivityEventSender || !m_eventSource.IsEnabled())
```

Comment:
Is there a way to simply disable (or not enable) the event source?

Reply:
https://github.com/Microsoft/referencesource/blob/master/mscorlib/system/diagnostics/eventing/eventsource.cs

Reply:
As far as I recall, we generally configure ETW providers at the cluster level, which would be the usual switch for an event source.

Reply:
But, based on the above source reference, if you can get a handle on the Event Source instance, you could probably get away with simply calling .Dispose() on it in a specific .NET App to disable it locally.

Reply:
I decice to do this because I dont want to disable the EventSource in case this can be used by other functions. It is safer and no harm in doing it through this setting.

Reply:
The EventSource is an ActivityEventSource that was created explicitly for emitting activities through this Activity Event Sender.
Disabling the Activity Event Sender and Disabling the Activity Event Source have the same net effect of preventing emission of activities through the ActivityEventSource.

Reply:
Another alternative may be to update SendActivityMetric in [AggregatedActivitiesEventSender](https://github.com/microsoft/Omex/blob/main/src/Activities/Internal/EventSource/AggregatedActivitiesEventSender.cs) to make sending through ActivityEventSender conditional on some overridable configuration.

Reply:
I've skimmed a few docs, it seems EventSource is enabled when there is a subscriber listening to it. I don't mind the IOption approach if we understand that it's a temporary hack. Proper solution is to make the entire data flow conditional, it will take quite some digging and refactoring.

Reply:
@K-Cully there is no easy way to disable EventSoruce that I can find, in terms of changing a setting or updating C# code. This way is the easiest and cleanest that can be done

Reply:
Resolving ActivityEventSource and calling .Dispose on it would work.
Either way it is a dirty hack, so I'm fine with whichever, so long as the follow-on removal is completed.